x-pack/filebeat/input/httpjson/input_test.go

```diff
@@ -1251,36 +1335,43 @@ var testCases = []struct {
   </xs:element>
 </xs:schema>
 `,
-		},
-		handler: defaultHandler(http.MethodGet, "", ""),
-		expected: []string{mapstr.M{
-			"order": map[string]interface{}{
-				"address": map[string]interface{}{
-					"address": "Beekplantsoen 594, 2 hoog, 6849 IG",
-					"city":    "Boekend",
-					"company": "Sydøstlige Gruppe",
-					"country": "Netherlands",
-					"name":    "Joord Lennart",
-				},
-				"item": []interface{}{
-					map[string]interface{}{
-						"cost":   99.95,
-						"name":   "Egil's Saga",
-						"note":   "Free Sample",
-						"number": 1,
-						"sent":   false,
+			},
+			handler: defaultHandler(http.MethodGet, "", ""),
+			expected: []string{mapstr.M{
+				"order": map[string]interface{}{
+					"address": map[string]interface{}{
+						"address": "Beekplantsoen 594, 2 hoog, 6849 IG",
+						"city":    "Boekend",
+						"company": "Sydøstlige Gruppe",
+						"country": "Netherlands",
+						"name":    "Joord Lennart",
+					},
+					"item": []interface{}{
+						map[string]interface{}{
+							"cost":   99.95,
+							"name":   "Egil's Saga",
+							"note":   "Free Sample",
+							"number": 1,
+							"sent":   false,
+						},
 					},
+					"noNamespaceSchemaLocation": "sales.xsd",
+					"orderid":                   "56733",
+					"sender":                    "Ástríðr Ragnar",
+					"xsi":                       "http://www.w3.org/2001/XMLSchema-instance",
 				},
-				"noNamespaceSchemaLocation": "sales.xsd",
-				"orderid":                   "56733",
-				"sender":                    "Ástríðr Ragnar",
-				"xsi":                       "http://www.w3.org/2001/XMLSchema-instance",
-			},
-		}.String()},
-	},
-}
+			}.String()},
+		},
+	}
+)
 
 func TestInput(t *testing.T) {
+	for i := 0; i < 2; i++ {
+		for j := 0; j < 2500; j++ {
+			expectedBulkData = append(expectedBulkData, fmt.Sprintf(`{ "text%d":%q }`, i+1, strings.Repeat(dummyText, 17)))
+		}
+	}
```

Comment:
```suggestion
	bulkData := make([]string, 5000)
	text := strings.Repeat(dummyText, 17)
	for i := range bulkData {
		bulkData[i] = fmt.Sprintf(`{ "text%d":%q }`, (i/2500)+1, text)
	}
```
Though, I'd put this in a helper and have it called below where it's used since we don't need it more than once.

Reply:
For now, I am keeping it there only. if we needed it anywhere else we will just move it to the helper function.

Reply:
I'd prefer it be moved. Suggest something like this
```diff
diff --git a/x-pack/filebeat/input/httpjson/input_test.go b/x-pack/filebeat/input/httpjson/input_test.go
index 2626ed1050..e8ca87c08e 100644
--- a/x-pack/filebeat/input/httpjson/input_test.go
+++ b/x-pack/filebeat/input/httpjson/input_test.go
@@ -36,6 +36,7 @@ var testCases = []struct {
        expectedFile string

        skipReason string
+       isLongTest bool
 }{
        {
                name:        "simple_GET_request",
@@ -813,6 +814,7 @@ var testCases = []struct {
        },
        {
                name:        "pagination_when_used_with_chaining_with_bulk_data",
+               isLongTest:  true,
                setupServer: newChainPaginationLoadTestServer(httptest.NewServer),
                baseConfig: map[string]interface{}{
                        "interval":       1,
@@ -836,6 +838,14 @@ var testCases = []struct {
                        },
                },
                handler: defaultHandler(http.MethodGet, "", ""),
+               expected: func() []string {
+                       bulkData := make([]string, 5000)
+                       text := strings.Repeat(dummyText, 17)
+                       for i := range bulkData {
+                               bulkData[i] = fmt.Sprintf(`{ "text%d":%q }`, (i/2500)+1, text)
+                       }
+                       return bulkData
+               }(),
        },
        {
                name: "replace_with_clause_and_first_response_object",
@@ -1362,11 +1372,6 @@ var testCases = []struct {
 const dummyText = "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum."

 func TestInput(t *testing.T) {
-       bulkData := make([]string, 5000)
-       text := strings.Repeat(dummyText, 17)
-       for i := range bulkData {
-               bulkData[i] = fmt.Sprintf(`{ "text%d":%q }`, (i/2500)+1, text)
-       }
        for _, test := range testCases {
                t.Run(test.name, func(t *testing.T) {
                        if test.skipReason != "" {
@@ -1401,12 +1406,9 @@ func TestInput(t *testing.T) {
                        g.Go(func() error {
                                return input.Run(ctx, chanClient)
                        })
-                       var timeoutDuration time.Duration
-                       if test.name == "pagination_when_used_with_chaining_with_bulk_data" {
+                       timeoutDuration := 10 * time.Second
+                       if test.isLongTest {
                                timeoutDuration = 2 * time.Minute
-                               test.expected = bulkData
-                       } else {
-                               timeoutDuration = 10 * time.Second
                        }
                        timeout := time.NewTimer(timeoutDuration)
                        t.Cleanup(func() { _ = timeout.Stop() })
```

Reply:
Okay, Got it. I will move it as you suggested.